src/providers/trusted_service/context/key_management.rs

```diff
@@ -35,38 +39,66 @@ impl Context {
         Ok(())
     }
 
+    /// Import a key given its attributes, ID, and key data.
+    ///
+    /// Lifetime flexibility is not supported: the `lifetime` parameter in the key
+    /// attributes is essentially ignored and replaced with `KeyLifetime::Persistent`.
+    ///
+    /// Key data must be in the format described by the PSA Crypto format.
     pub fn import_key(
         &self,
         key_attrs: Attributes,
         id: u32,
         key_data: &[u8],
     ) -> Result<(), ResponseStatus> {
+        info!("Handling ImportKey request");
+        let mut data = key_data.to_vec();
         let mut import_req = ImportKeyIn {
             attributes: Some(KeyAttributes {
-                r#type: u16::try_from(key_attrs.key_type)? as u32,
-                key_bits: key_attrs.bits.try_into()?,
+                r#type: u16::try_from(key_attrs.key_type).map_err(|e| {
+                    data.zeroize();
+                    e
+                })? as u32,
+                key_bits: key_attrs.bits.try_into().map_err(|e| {
+                    data.zeroize();
+                    e
+                })?,
                 lifetime: KeyLifetime::Persistent as u32,
                 id,
                 policy: Some(KeyPolicy {
-                    usage: key_attrs.policy.usage_flags.try_into()?,
-                    alg: key_attrs.policy.permitted_algorithms.try_into()?,
+                    usage: key_attrs.policy.usage_flags.into(),
+                    alg: key_attrs
+                        .policy
+                        .permitted_algorithms
+                        .try_into()
+                        .map_err(|e| {
+                            data.zeroize();
+                            e
+                        })?,
                 }),
             }),
-            data: key_data.to_vec(),
+            data,
```

Comment:
If you just transform it into a vector at this point then you don't need to zeroize it before when it fails right?

Reply:
I'm unsure what the compiler does - I don't think it necessarily does all those conversions in order. Was hoping that constructing `data` before would make it do that for sure, and hopefully all the conversions happen after that

Reply:
Ah yes maybe. In that case my other suggestion would be to wrap the vector into a `Zeroizing` so that it's cleared when dropped wihtout having to zeroize on every error

Reply:
That's not really helping, though, because then I have to "convert" from that `Zeroizing` into a `Vec` :] the thing I'm populating here is a protobuf type, I don't have a choice as to what I fill it with.

Reply:
ah ok fair enough then